smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/AbstractMultiUserChatIntegrationTest.java

```diff
@@ -55,10 +56,34 @@ public AbstractMultiUserChatIntegrationTest(SmackIntegrationTestEnvironment envi
 
         List<DomainBareJid> services = mucManagerOne.getMucServiceDomains();
         if (services.isEmpty()) {
-            throw new TestNotPossibleException("No MUC (XEP-45) service found");
+            throw new TestNotPossibleException("No MUC (XEP-0045) service found");
         }
 
-        mucService = services.get(0);
+        DomainBareJid needle = null;
+        for (final DomainBareJid service : services) {
+            MultiUserChat multiUserChat = null;
+            try {
+                final String roomNameLocal = String.join("-", "smack-inttest-abstract", testRunId, StringUtils.insecureRandomString(6));
+                final EntityBareJid mucAddress = JidCreate.entityBareFrom(Localpart.from(roomNameLocal), service.getDomain());
+                multiUserChat = mucManagerOne.getMultiUserChat(mucAddress);
+                createMuc(multiUserChat, Resourcepart.from("test"));
+                needle = service;
+                break;
+            } catch (XMPPException.XMPPErrorException | MultiUserChatException.MucAlreadyJoinedException |
+                     MultiUserChatException.MissingMucCreationAcknowledgeException | XmppStringprepException |
+                     MultiUserChatException.NotAMucServiceException e) {
+                LOGGER.log(Level.FINER, "Error creating test MUC", e);
```

Comment:
Writing the comment "below" in MultiUserChatLowLevelIntegrationTest made me realize that we can and should be more be more precise here. Only catch `XMPPErrorException` here, and only continue the loop, i.e., do not rethrow if the error condition is `not-allowed`. I think this is the common condition that is returned if the user is not allowed to create MUC. I wonder if, depending on the exact reason why a user is not allowed to create a room, other errror conditions could also be returned. XEP-0045 does not seem to talk about it, but it feels like a possibility.